### ISSUE TITLE
architecture dependand files installed in /usr/share

### ISSUE THREAD

**Issue by arekm** [ruz/Alien-LibGumbo#2]
/usr/share is meant for architecture independant files (http://www.pathname.com/fhs/2.2/fhs-4.11.html).

Unfortunately Alien-LibGumbo installs architecture dependand files there, example:

```
        /usr/share/perl5/vendor_perl/auto/share/dist/Alien-LibGumbo/lib/libgumbo.a
        /usr/share/perl5/vendor_perl/auto/share/dist/Alien-LibGumbo/lib/libgumbo.la
        /usr/share/perl5/vendor_perl/auto/share/dist/Alien-LibGumbo/lib/libgumbo.so
        /usr/share/perl5/vendor_perl/auto/share/dist/Alien-LibGumbo/lib/libgumbo.so.1
        /usr/share/perl5/vendor_perl/auto/share/dist/Alien-LibGumbo/lib/libgumbo.so.1.0.0
```

These should be installed somewhere in /usr/lib64/perl5/vendor_perl/5.18.0/x86_64-pld-linux-thread-multi/ (on my distro) aka in /usr/lib64/...

**Comment by plicease** [comment]
This is actually an issue with `Alien::Base`, not `Alien::LibGumbo`.  I attempted to get AB based dists to install in arch directories by default, but could not secure consensus on this.  See GitHub issues:

https://github.com/Perl5-Alien/Alien-Base/issues/118
https://github.com/Perl5-Alien/Alien-Base/issues/119

TL;DR: there is an ALIEN_ARCH environment variable that you can set to change the default behavior.  This is especially useful for people working in non-homogeneous environments (like AFS) or for linux distribution vendors.  However, if you are a distro vendor packager, you should consider using the system library (in which case you will not see arch dependent files in the share directory) instead #3 should make this possible.

For a lot more detail please see my blog on the subject:

http://blogs.perl.org/users/graham_ollis/2015/07/alienbase-system-integrators-vs-cpan-authors.html

If the use of ALIEN_ARCH environment variable seems onerous, please do consider commenting on one of the AB issues above, as I may be able to gain consensus for changing the default if more people ask for it.

**Comment by ruz** [comment]
Reopen it if Alien-Base fixed this and I have to do something with the module to get it in sync and fix this.

**Comment by plicease** [comment]
Issue is not with Alien::Base, but with the installer.

The older installer Alien::Base::ModuleBuild installs into lib not arch. I personally did not agree with this approach, but the consensus was to keep the old behavior. Once again, this is not likely to change without user advocacy.

The newer installer alienfile/Alien::Build installs to the arch directory by default (the argument being that most Aliens will be arch dependent, and the newer installer did not have to be backward compatible with AB::MB).  There are other advantages to alienfile/Alien::Build like fewer dependencies and the code base is being actively developed rather than in maintenance mode.
